src/clzones.cpp

```diff
@@ -732,6 +832,188 @@ void zone_data::set_temporary_disabled( const bool enabled_arg )
     temporarily_disabled = enabled_arg;
 }
 
+static field_type_str_id field_of( zone_data *zone )
+{
+    zone_type_id type = zone->get_type();
+
+    if( type == zone_type_AUTO_DRINK ) {
+        return fd_auto_drink_zone;
```

Comment:
Please make this a json-defined value of `LOOT_ZONE` instead of hardcoding it.

Reply:
I'm trying this, which was easier than I'd expected, but I've hit a snag: The hard coded zones aren't found when looking for zones: `zone_type.get_all()`, presumably because they're actually only created locally within the UI, and so never get "registered", which I think happens during the JSON reading process.
Thus, when trying to toggle a Basecamp Storage zone (for instance) none is found because the zone_type class doesn't know about it.

I think a good approach would be to define these zones in JSON as well. That would move the UI texts for them from code into JSON, but would also break the code functionality tied to them if someone was to change their identifiers. However, I can imagine it might also change the order in which they appear in the zone type selection list (which I don't think would be a big issue).

Do you have any objections to this approach (e.g. "it's a bad idea because..."), or other thoughts?